qualtran/surface_code/t_factory.py

```diff
@@ -0,0 +1,52 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import math
+
+from attrs import field, frozen
+
+from qualtran.surface_code.algorithm_summary import AlgorithmSummary
+from qualtran.surface_code.magic_state_factory import MagicStateFactory
+
+
+@frozen
+class TFactory(MagicStateFactory):
+    """A magic state factory for T states.
+
+    Attributes:
+        num_qubits: Number of qubits used by the factory.
```

Comment:
physical or logical?

Reply:
In another comment you say

> none of them is a phsical parameter.

?

Reply:
well it's not a physical parameter as in time or error. It's just the number of qubits this factory. The factory described in the paper works with physical qubits